web/controllers/api/docs_controller.ex

```diff
@@ -199,4 +201,29 @@ defmodule HexWeb.API.DocsController do
     # TODO: Send mails
     HexWeb.Utils.task(task, fn -> nil end, fn _ -> nil end)
   end
+
+  defp publish_sitemap do
+      # query for has docs packages
+      packages = HexWeb.Package.all_has_docs |> HexWeb.Repo.all
+
+      # build sitemap
+      template = """
+<?xml version="1.0" encoding="utf-8"?>
+<urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9"
+ xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+ xsi:schemaLocation="http://www.sitemaps.org/schemas/sitemap/0.9 http://www.sitemaps.org/schemas/sitemap/0.9/sitemap.xsd">
+ <%= for package <- packages do %><url>
+     <loc>http://hexdocs.pm/<%= package.name %></loc>
+     <lastmod><%=  Ecto.DateTime.to_iso8601(package.updated_at) %></lastmod>
```

Comment:
I think we need to add a new datetime field to package that is set to when the main (non-versioned) docs were last updated.

The reason for this is because in the future we will add some logic that will conditionally update the main docs. For example pre-releases should not automatically replace main docs.